maingate/src/range.rs

```diff
@@ -283,6 +278,13 @@ impl<F: FieldExt> RangeChip<F> {
             overflow_tables,
         }
     }
+
+    fn bases(&self, limb_bit_len: usize) -> &[F] {
+        self.bases
+            .get(&limb_bit_len)
+            .unwrap_or_else(|| panic!("composition table is not set, bit lenght: {}", limb_bit_len))
```

Comment:
Can't you simply use `.expect()`? Seems simpler and serves the same exact purpose.

Reply:
Since the error string has to be formatted `clippy` will throw the warning  `clippy::expect_fun_call` if we use `expect()`.  It suggests to use the lazily evaluated `unwrap_or_else` instead.

Reply:
Clippy just passed for me with this updated in #40  :/
Basically you can use `format!` and the macro code gets expanded. Which should be as good as lazily throwing the panic message.

Reply:
It passes because there is no `clippy` check in the CI. Maybe we should add it.